Src/PerformanceCollector/Shared/QuickPulseTelemetryProcessor.cs

```diff
@@ -157,13 +165,78 @@ public void Process(ITelemetry telemetry)
             }
             finally
             {
-                if (letItemThrough)
+                if (letTelemetryThrough)
                 {
                     this.Next.Process(telemetry);
                 }
             }
         }
 
+        private static ITelemetryDocument ConvertTelemetryToTelemetryDocument(
+            RequestTelemetry telemetryAsRequest,
+            DependencyTelemetry telemetryAsDependency,
+            ExceptionTelemetry telemetryAsException)
```

Comment:
I don't mean for you to have to do type conversion, but shouldn't the conditional statement "if (telemetry is RequestTelemetry)" execute almost as well as "if (telemetryAsRequest != null)"?